src/shapes/object.class.ts

```diff
@@ -1402,13 +1408,13 @@ export class FabricObject extends ObjectGeometry {
    * @param {CanvasRenderingContext2D} ctx Context to set the dash line on
    * @param {Array} dashArray array representing dashes
    */
-  _setLineDash(ctx, dashArray) {
+  _setLineDash(ctx: CanvasRenderingContext2D, dashArray?: number[] | null) {
     if (!dashArray || dashArray.length === 0) {
       return;
     }
-    // Spec requires the concatenation of two copies the dash list when the number of elements is odd
+    // Spec requires the concatenation of two copies of the dash array when the number of elements is odd
     if (1 & dashArray.length) {
-      dashArray.push.apply(dashArray, dashArray);
+      dashArray.push(...dashArray);
```

Comment:
Should we mutate the arg?
Or should we instead
```ts
ctx.setLineDash(1 & dashArray.length ? [...dashArray, ...dashArray]: dashArray);
```

Reply:
from the tests I see this is intentional

Reply:
Yeah in theory we shouldn't and we should modify it every single time.
But also eventually we can do what we want with property assigned to an instance?
The intention was to not redo it every time